src/sentry/models/savedsearch.py

```diff
@@ -32,28 +34,44 @@
 DEFAULT_SAVED_SEARCH_QUERIES = set(search['query'] for search in DEFAULT_SAVED_SEARCHES)
 
 
+class SavedSearchType(IntEnum):
+    ISSUE = 0
+    EVENT = 1
+
+
 class SavedSearch(Model):
     """
     A saved search query.
     """
     __core__ = True
-
+    # TODO: Remove this column and rows where it's not null once we've
+    # completely removed Sentry 9
     project = FlexibleForeignKey('sentry.Project', null=True)
+    organization = FlexibleForeignKey('sentry.Organization', null=True)
+    type = models.PositiveSmallIntegerField(default=SavedSearchType.ISSUE.value, null=True)
     name = models.CharField(max_length=128)
     query = models.TextField()
     date_added = models.DateTimeField(default=timezone.now)
+    # TODO: Remove this column once we've completely removed Sentry 9
     is_default = models.BooleanField(default=False)
     is_global = models.NullBooleanField(null=True, default=False, db_index=True)
     owner = FlexibleForeignKey('sentry.User', null=True)
```

Comment:
If we re-use the current owner column could that complicate users moving between saved search implementations? If a user make a pinned search, disables the feature should their org saved searches become inaccessible?

Reply:
I don't quite follow, could you elaborate on this case more?

Reply:
Disregard my comment. If we ensure that project saved searches can never get an org id and vice versa then we won't have any issues.